lib/IRGen/GenCast.cpp

```diff
@@ -210,7 +210,8 @@ llvm::Value *irgen::emitClassDowncast(IRGenFunction &IGF, llvm::Value *from,
 
   // If the destination type is known to have a Swift-compatible
   // implementation, use the most specific entrypoint.
-  if (destClass && destClass->hasKnownSwiftImplementation()) {
+  if ((destClass && destClass->hasKnownSwiftImplementation()) ||
+      IGF.IGM.Context.LangOpts.hasFeature(Feature::Embedded)) {
```

Comment:
Is it ever the case that `(destClass && destClass->hasKnownSwiftImplementation())` would not be true in `Embedded` mode? Could this be an `assert(!hasFeature(Embedded))` assertion on the else path?

Reply:
I'm not sure